aten/src/ATen/native/Indexing.cpp

```diff
@@ -569,4 +569,20 @@ Tensor _gather_sparse_backward(const Tensor& self, int64_t dim, const Tensor& in
     return at::_sparse_coo_tensor_unsafe(sparse_ind, grad.reshape(-1), self.sizes());
 }
 
+std::vector<Tensor> nonzero_tuple(const Tensor& self) {
+  // special case scalar for compatibility with numpy:
+  //
+  // >>> np.array(5).nonzero()
+  // (array([0]),)
+  // >>> np.array(0).nonzero()
+  // (array([], dtype=int64),)
+  if (self.dim() == 0) {
+    if (at::is_nonzero(self)) {
```

Comment:
I think you can just unsqueeze in this case to get the correct behavior.

Reply:
Is this what you mean? It doesn't give the correct behavior.

```python
>>> torch.tensor(5).unsqueeze(0)
tensor([5])
>>> np.array(5).nonzero()
(array([0]),)
```

Reply:
I meant `unsqueeze` before you call into nonzero.

Reply:
(y)